x/tx/decode/decode.go

```diff
@@ -138,14 +141,18 @@ func (dtx *DecodedTx) Hash() [32]byte {
 	return sha256.Sum256(bz)
 }
 
-func (dtx *DecodedTx) GetGasLimit() uint64 {
+func (dtx DecodedTx) GetGasLimit() uint64 {
 	return dtx.Tx.AuthInfo.Fee.GasLimit
 }
 
-func (dtx *DecodedTx) GetMessages() []proto.Message {
+func (dtx DecodedTx) GetMessages() []proto.Message {
 	return dtx.Messages
 }
 
-func (dtx *DecodedTx) GetSenders() [][]byte {
+func (dtx DecodedTx) GetSenders() [][]byte {
 	return dtx.Signers
 }
+
+func (dtx DecodedTx) Bytes() []byte {
```

Comment:
```suggestion
func (dtx *DecodedTx) Bytes() []byte {
```

and revert the above method receivers to points too, unless I'm missing something?

Reply:
runtime.ServerTxToSDKTx function expects an argument that implements the transaction.Tx interface, and the transaction.Tx interface has a non-pointer receiver method. Otherwise, it leads to type mismatch.